swift/ql/lib/upgrades/c0db61944f46ba5507f207ec2b1cff77ad0529a1/pattern_types.ql

```diff
@@ -0,0 +1,9 @@
+class Element extends @element {
+    string toString() {
+        none()
+    }
+}
+
+from Element p, Element t
+where none()
+select p, t
```

Comment:
What does this script do? Is it simple removing all the elements?

Reply:
I think `delete` in the `upgrade.properties` does this, see https://github.com/github/codeql/blob/main/docs/prepare-db-upgrade.md .

Reply:
... but that's what you've done in the downgrade script. Are we just trying to synthesize an empty table here?

Reply:
I'm not sure you need to.  I definitely don't consider myself an expert though.